diff --git a/source/plugins/languages/analyzer/recent.mjs b/source/plugins/languages/analyzer/recent.mjs
--- a/source/plugins/languages/analyzer/recent.mjs
+++ b/source/plugins/languages/analyzer/recent.mjs
@@ -67,6 +67,6 @@ export class RecentAnalyzer extends Analyzer {
       ...await Promise.allSettled(
         commits
-          .flatMap(({payload}) => payload.commits)
+          .flatMap(({payload}) => payload.commits ?? [])
           .filter(({committer}) => filters.text(committer?.email, this.authoring, {debug: false}))
           .map(commit => commit.url)
           .map(async commit => (await this.rest.request(commit)).data),
